dataprofiler/profilers/profile_builder.py

```diff
@@ -1884,27 +1950,54 @@ def report(self, report_options: dict = None) -> dict:
 
         return _prepare_report(report, output_format, omit_keys)
 
-    def _get_unique_row_ratio(self) -> float:
+    def _get_unique_row_ratio(self) -> float | None:
         """Return unique row ratio."""
+        if (
+            not self.options.row_statistics.is_enabled
+            or not self.options.row_statistics.unique_count.is_enabled
+        ):
+            return None
+
         if self.total_samples:
-            return len(self.hashed_row_dict) / self.total_samples
+            if isinstance(self.hashed_row_object, dict):
+                return len(self.hashed_row_object) / self.total_samples
+            elif isinstance(self.hashed_row_object, HyperLogLog):
+                return int(self.hashed_row_object.cardinality()) / self.total_samples
         return 0
 
-    def _get_row_is_null_ratio(self) -> float:
+    def _get_row_is_null_ratio(self) -> float | None:
         """Return whether row is null ratio."""
+        if not self.options.row_statistics.is_enabled:
```

Comment:
here is the addition that makes row stats enabled for null calcs to return non-null values